pkg/storage/mongo/cataloger.go

```diff
@@ -3,34 +3,39 @@ package mongo
 import (
 	"context"
 
-	"github.com/globalsign/mgo/bson"
 	"github.com/gomods/athens/pkg/errors"
 	"github.com/gomods/athens/pkg/paths"
 	"github.com/gomods/athens/pkg/storage"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
 // Catalog implements the (./pkg/storage).Cataloger interface
 // It returns a list of modules and versions contained in the storage
 func (s *ModuleStore) Catalog(ctx context.Context, token string, pageSize int) ([]paths.AllPathParams, string, error) {
 	const op errors.Op = "mongo.Catalog"
-	q := bson.M{}
+	q := bson.E{}
 	if token != "" {
-		q = bson.M{"_id": bson.M{"$gt": bson.ObjectIdHex(token)}}
+		q = bson.E{Key: "_id", Value: bson.M{"$gt": token}}
 	}
 
-	fields := bson.M{"module": 1, "version": 1}
+	projection := bson.M{"module": 1, "version": 1}
+
+	c := s.client.Database(s.db).Collection(s.coll)
 
-	c := s.s.DB(s.d).C(s.c)
 	modules := make([]storage.Module, 0)
-	err := c.Find(q).
-		Select(fields).
-		Sort("_id").
-		Limit(pageSize).
-		All(&modules)
+	cursor, err := c.Find(context.Background(), q, &options.FindOptions{Projection: projection})
 
 	if err != nil {
 		return nil, "", errors.E(op, err)
 	}
+	for cursor.Next(context.Background()) {
+		var module storage.Module
+		elem := &bson.D{}
+		if err := cursor.Decode(module); err == nil {
```

Comment:
I don't think we should ignore errors. It's better to fail than to return incomplete data.
Could you use `multierror` like here https://github.com/gomods/athens/blob/d29efb7d61c34b988fe4428772fd8089bc6e1f88/pkg/storage/minio/saver.go#L104 ?